pkg/crc/machine/kubeconfig.go

```diff
@@ -242,33 +242,42 @@ func contains(arr []string, str string) bool {
 	return false
 }
 
-func mergeKubeConfigFile(kubeConfigFile string) error {
-	globalConfigPath, kubeConf1, err := getGlobalKubeConfig()
-	if err != nil {
-		return err
+func mergeKubeConfigFile(kubeConfigFile, globalConfigFile string) error {
+	var kubeConf1 *api.Config
+	var err error
+	if globalConfigFile != "" {
+		kubeConf1, err = clientcmd.LoadFromFile(globalConfigFile)
+		if err != nil {
+			return err
+		}
+	} else {
+		_, kubeConf1, err = getGlobalKubeConfig()
+		if err != nil {
+			return err
+		}
 	}
 	kubeConf2, err := clientcmd.LoadFromFile(kubeConfigFile)
 	if err != nil {
 		return err
 	}
 	// Merge the kubeConf2 to globalConfig
-	mergedConfig, err := clientcmd.NewDefaultClientConfig(*kubeConf1, &clientcmd.ConfigOverrides{}).ConfigAccess().GetStartingConfig()
-	if err != nil {
-		return err
-	}
 
 	for name, cluster := range kubeConf2.Clusters {
-		mergedConfig.Clusters[name] = cluster
+		kubeConf1.Clusters[name] = cluster
 	}
 
 	for name, authInfo := range kubeConf2.AuthInfos {
-		mergedConfig.AuthInfos[name] = authInfo
+		kubeConf1.AuthInfos[name] = authInfo
 	}
 
 	for name, context := range kubeConf2.Contexts {
-		mergedConfig.Contexts[name] = context
+		kubeConf1.Contexts[name] = context
 	}
 
-	mergedConfig.CurrentContext = kubeConf2.CurrentContext
-	return clientcmd.WriteToFile(*mergedConfig, globalConfigPath)
+	configAccess := clientcmd.NewDefaultPathOptions()
+	configAccess.GlobalFile = globalConfigFile
+	configAccess.LoadingRules.ExplicitPath = globalConfigFile
```

Comment:
Looking at https://github.com/kubernetes/client-go/blob/86d49e7265f07676cb39f342595a858b032112de/tools/clientcmd/config.go#L145-L158 , the defaults for `GlobalFile` and `LoadingRules` look useful when `globalKubeConfigFile` is `""`? In other words, maybe this should be changed to:
```
if globalConfigFile != "" {
    configAccess.GlobalFile = globalConfigFile
    configAccess.LoadingRules.ExplicitPath = globalConfigFile
}
```
? Or is this also required when modifying the global kubeconfig file?

Reply:
I just updated the code again. I found that the `.LocationOfOrigin` field was being given priority in choosing the output file so I changed it to the global kubeconfig file.